prow/artifact-uploader/controller.go

```diff
@@ -45,6 +47,10 @@ const (
 	ContainerLogDir = "logs"
 )
 
+type prowJobClient interface {
```

Comment:
Nit: Do we get anything from defining this interface? The client has two implementations, the real one and the fake one. Both of them implement the full `v1.ProwJobInterface`

Reply:
I simply saw this pattern elsewhere and thought it was a convention. I have removed the use of this intermediate interface.